### ISSUE TITLE
HTTPS_PROXY environment variable is not always honored when downloading assets from Ultralytics Platform

### ISSUE THREAD

**Issue by muzhikov** [ultralytics/ultralytics#25205]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Other

### Bug

HTTPS_PROXY environment variable is honored during model download and dataset download (accessing https://storage.googleapis.com/...), but **ignored** during dataset images dowload (accessing https://cdn.ul.run/.../...jpg)

### Environment

Ultralytics 8.4.90 🚀 Python-3.14.2 torch-2.12.1 CPU (Apple M2 Max)
Setup complete ✅ (12 CPUs, 32.0 GB RAM, 743.5/926.4 GB disk)

OS                     macOS-15.5-arm64-arm-64bit-Mach-O
Environment            Darwin
Python                 3.14.2
Install                pip
Path                   /Users/andrey/Repository/6k2/ballot-recognition/.venv/lib/python3.14/site-packages/ultralytics
RAM                    32.00 GB
Disk                   743.5/926.4 GB
CPU                    Apple M2 Max
CPU count              12
GPU                    None
GPU count              None
CUDA                   None

numpy                  ✅ 2.5.1>=1.23.0
matplotlib             ✅ 3.11.0>=3.3.0
opencv-python          ✅ 5.0.0.93!=4.13.0.90,>=4.6.0
pillow                 ✅ 12.3.0>=7.1.2
pyyaml                 ✅ 6.0.3>=5.3.1
requests               ✅ 2.34.2>=2.23.0
torch                  ✅ 2.12.1>=1.8.0
torch                  ✅ 2.12.1!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision            ✅ 0.27.1>=0.9.0
psutil                 ✅ 7.2.2>=5.8.0
polars                 ✅ 1.42.1>=0.20.0
nvidia-ml-py           ✅ 13.610.43>=12.0.0
ultralytics-thop       ✅ 2.0.20>=2.0.18

### 1. Run proxy server 
Run http proxy server on my_host:my_port

### 2. Export HTTPS_PROXY in console
`export HTTPS_PROXY='http://my_host:my_port'`

### 3. Execute Python code and check proxy server logs:
```
from ultralytics import YOLO

model = YOLO("ul://my_account/my_project/my_model")    
# ^^^ see access to 'https://storage.googleapis.com/...ultralytics/...' in proxy server logs

model.val( data="ul://my_account/datasets/my_dataset" )
# ^^^ see access to 'https://storage.googleapis.com/...ultralytics/...' in proxy server logs
# BUT NO access to 'https://cdn.ul.run/.../....jpg' since jpg files are served directly, and NOT via HTTPS_PROXY
```

### Additional

The problem is in this line:
https://github.com/ultralytics/ultralytics/blob/78f38f9a8171f9d297b783debae582593397f795/ultralytics/data/converter.py#L1018

Since aiohttp ignores environment variables unless they are explicitly trusted. It should be:
`async with aiohttp.ClientSession(trust_env=True) as session:`

### Are you willing to submit a PR?

- [x] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @muzhikov, thank you for the detailed report and for including an MRE 🚀! This is an automated response; an Ultralytics engineer will also assist you soon.

We recommend visiting the [Docs](https://docs.ultralytics.com/) for new users, including [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and answers to common questions.

Your proxy configuration, environment details, reproduction steps, and proposed fix are helpful. To assist with investigation, please also include:
- The complete traceback or error output, if any
- The exact dataset and model download commands that reproduce the behavior
- Proxy server logs showing requests to `storage.googleapis.com` versus `cdn.ul.run`
- Confirmation that the issue persists after upgrading to the latest `ultralytics` package

If this is a custom training ❓ question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips).

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.
